packages/date-picker/test/form-input.test.js

```diff
@@ -162,51 +168,176 @@ describe('form input', () => {
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should pass validation with a value = min', () => {
+      datePicker.value = '2010-01-01';
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should fail validation when the entered value < min', async () => {
+      setInputValue(datePicker, '1/1/2000');
+      await close(datePicker);
+      expect(datePicker.checkValidity()).to.be.false;
+      expect(datePicker.validate()).to.be.false;
+    });
+
+    it('should pass validation when the entered value > min', async () => {
+      setInputValue(datePicker, '1/1/2022');
+      await close(datePicker);
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should pass validation when the entered value = min', async () => {
+      setInputValue(datePicker, '1/1/2022');
+      await close(datePicker);
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+  });
+
+  describe('max', () => {
+    beforeEach(() => {
+      datePicker = fixtureSync(`<vaadin-date-picker max="2010-01-01"></vaadin-date-picker>`);
+    });
+
+    it('should pass validation without value', () => {
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should fail validation with a value > max', () => {
+      datePicker.value = '2020-01-01';
+      expect(datePicker.checkValidity()).to.be.false;
+      expect(datePicker.validate()).to.be.false;
+    });
+
+    it('should pass validation with a value < max', () => {
+      datePicker.value = '2000-01-01';
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
+    });
+
+    it('should pass validation with a value = max', () => {
+      datePicker.value = '2010-01-01';
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
     });
 
-    it('should not be invalid without user interactions', () => {
-      expect(datepicker.invalid).to.be.false;
+    it('should fail validation when the entered value > max', async () => {
+      setInputValue(datePicker, '1/1/2022');
+      await close(datePicker);
+      expect(datePicker.checkValidity()).to.be.false;
+      expect(datePicker.validate()).to.be.false;
     });
 
-    it('should be invalid after validate() if value is not set', () => {
-      datepicker.validate();
-      expect(datepicker.invalid).to.be.true;
+    it('should pass validation when the entered value < max', async () => {
+      setInputValue(datePicker, '1/1/2000');
+      await close(datePicker);
+      expect(datePicker.checkValidity()).to.be.true;
+      expect(datePicker.validate()).to.be.true;
     });
 
-    it('should focus on required indicator click', () => {
```

Comment:
**note:** Moved to [`basic.test.js`](https://github.com/vaadin/web-components/pull/4163/files?w=0#diff-3ba8bda46eb1d29828d621d853b039ba6d46039afdccc4b25fbb1adcae4aeb26R742).